sympy/functions/combinatorial/factorials.py

```diff
@@ -339,13 +339,18 @@ def eval(cls, arg):
                 if arg.is_even:
                     k = arg / 2
                     return 2 ** k * factorial(k)
-
-                return factorial(arg) / factorial2(arg - 1)
+                if arg.is_odd:
+                    return factorial(arg) / factorial2(arg - 1)
+                raise ValueError("argument must be nonnegative integer or odd integer")
 
             if arg.is_even:
                 raise ValueError("argument must be nonnegative or odd")
 
-            return arg * (S.NegativeOne) ** ((1 - arg) / 2) / factorial2(-arg)
+            if arg.is_integer:
+                return arg * (S.NegativeOne) ** ((1 - arg) / 2) / factorial2(-arg)
+            raise ValueError("argument must be nonnegative integer or odd integer")
+        else:
+            raise ValueError("argument must be nonnegative integer or odd integer")
```

Comment:
This line should not be added. It will be executed, for example, when arg is a symbol. This is wrong because the symbol may be intended to denote an integer. Instead of raising ValueError the method should simply return (with implied None).